patterns/Singleton/README.md

```diff
@@ -0,0 +1,57 @@
+# Singleton
+
+## TL;DR
+
+생성자가 여러 번 호출이 되어도 실제 생성되는 객체는 하나이고 최초 생성 이후에는 호출이 된 생성자는 최초의 생성자가 만든 객체를 리턴하는 방법. 즉, 전체 시스템 통틀어서 싱글톤 객체는 유일하며, 전역변수처럼 취급이 될 수 있다.
+
+## Problem
+
+로그를 기록하는 클래스를 정의했다고 가정하자. 각 레이어에서 이 클래스를 이용하여 같은 역할을 하는 객체를 생성한다면, 같은 기능을 하는 중복된 객체가 생성이 되므로 효율이 좋지 않다. 또한 코드 재활용이라는 기준으로 볼 때 적합하지 않다.
```

Comment:
중복 객체 생성 방지와 코드 재활용이 singleton 패턴이 해결하려는 근본적인 문제일까요? 제한된 리소스에 대한 액세스 패턴을 제어하기 위한 문제에 대한 내용도 함께 포함되면 좋을 것 같습니다.

> gof
> Motivation
> It’s important for some classes to have exactly one instance. Although there can be many printers in a system, there should be only one printer spooler. There should be only one file system and one window manager. A digital filter will have one A/D converter. An accounting system will be dedicated to serving one company.

> dive into design patterns
> * Why would any- one want to control how many instances a class has? The most common reason for this is to control access to some shared resource—for example, a database or a file.
>  * Remember those global variables that you (all right, me) used to store some essential objects? While they’re very handy, they’re also very unsafe since any code can potentially overwrite the contents of those variables and crash the app.

Reply:
Problem에서 가정하는 상황과 examples의 상황이 일치하면 좋을 것 같아요
example마다 문제상황이 다르면 파악도 어려워지고 언어별 비교도 힘들어질 것 같습니다
그래서 각 언어별 sample과 문서의 problem이 일치하면 좋을 것 같네요